src/Zip/ZipFile.SaveSelfExtractor.cs

```diff
@@ -757,9 +767,14 @@ private void _SaveSfxStub(string exeToGenerate, SelfExtractorSaveOptions options
                 // get the Ionic.Zip assembly
                 Assembly a1 = typeof(ZipFile).Assembly;
 
-                using (var csharp = new Microsoft.CSharp.CSharpCodeProvider
-                       (new Dictionary<string,string>() { { "CompilerVersion", "v2.0" } })) {
+                var providerOptions = new Dictionary<string, string>();
+                if (!string.IsNullOrEmpty(options.CompilerVersion))

```

Comment:
If version is not specified it will use the highest available (v4.0) as all of them are inplace updates.

This was not like this previously. It was introduced latter. I will look up the commit if you'd like to.

Reply:
If you take a look at this commit and the change it was introduced in, then you see this option shouldn't even have to be added before. Or I have missed something :)

https://github.com/haf/DotNetZip.Semverd/blame/c54a4a4ec4baa1859004dc666c4e58e8a8150ef7/src/Zip/ZipFile.SaveSelfExtractor.cs

Reply:
Understood, thanks for clarification.
My thoughts - it would indeed be better to use the highest available .NET framework version here, especially given the fact that .NET 2.0 is history now and isn't installed by default in recent versions of Windows.
My only concern now is if it actually works or not with default setting (highest framework version) - have you checked it any recent versions of Windows, like Win10?

Reply:
We've tested with Windows Server 2012 and Win 10 Enterprise. They both compiled with 4.0 without giving the options.